src/hooks/useTxDetails/useTxDetails.ts

```diff
@@ -70,29 +71,54 @@ export const getTxType = (tx: Tx, transfers: Transfer[]): TxType => {
 }
 
 export const getTransfers = (
-  transfers: TxTransfer[],
+  tx: Tx,
   assets: AssetsById,
   marketData: Record<AssetId, MarketData | undefined>,
 ): Transfer[] => {
-  return transfers.reduce<Transfer[]>((prev, transfer) => {
-    const asset = assets[transfer.assetId]
+  return tx.transfers.reduce<Transfer[]>((prev, transfer) => {
+    const asset = (() => {
+      const { assetNamespace } = fromAssetId(transfer.assetId)
+      switch (assetNamespace) {
+        case 'erc721':
+        case 'erc1155':
+        case 'bep721':
+        case 'bep1155': {
+          const icon = (() => {
+            if (!tx.data || !transfer.id || !('mediaById' in tx.data)) return
+            return tx.data.mediaById[transfer.id]?.url
+          })()
+
+          const asset = makeAsset({
+            assetId: transfer.assetId,
+            id: transfer.id,
+            symbol: transfer.token?.symbol ?? 'N/A',
+            name: transfer.token?.name ?? 'Unknown',
+            precision: 0,
+            icon,
+          })
+
```

Comment:
I was attempting to upsert the assets into the store here, but it was causing re render issues, not sure if there would be a better way to do this. The only other place this could occur would be in the txHistory slice when fetching tx history, so maybe that would allow upserting